expyfun/io/_parse.py

```diff
@@ -73,3 +99,110 @@ def read_tab(fname, group_start='trial_id', group_end='trial_ok'):
+    tr : list of TrackerUD or TrackerBinom
+        The tracker objects with all responses such that they are in their
+        stopped state (as long as the trackers were allowed to stop during
+        the generation of the file.)
+    """
+    from ..stimuli import TrackerUD, TrackerBinom
+    # read in raw data
+    raw = read_tab_raw(fname)
+
+    # find tracker_identify and make list of IDs
+    tracker_idx = np.where([r[1] == 'tracker_identify' for r in raw])[0]
+    if len(tracker_idx) == 0:
+        raise ValueError('There are no Trackers in this file.')
+    tr = []
+    for ii in tracker_idx:
+        tracker_id = ast.literal_eval(raw[ii][2])['tracker_id']
+        tracker_type = ast.literal_eval(raw[ii][2])['tracker_type']
+        # find tracker_ID_init lines and get dict
+        init_str = 'tracker_' + str(tracker_id) + '_init'
+        tracker_dict_idx = np.where([r[1] == init_str for r in raw])[0][0]
+        tracker_dict = json.loads(raw[tracker_dict_idx][2])
+        if tracker_type == 'TrackerUD':
+            tr.append(TrackerUD(**tracker_dict))
+        else:
+            tr.append(TrackerBinom(**tracker_dict))
+        tr[-1]._tracker_id = tracker_id  # make sure tracker has original ID
+        stop_str = 'tracker_' + str(tracker_id) + '_stop'
+        tracker_stop_idx = np.where([r[1] == stop_str for r in raw])[0]
+        if len(tracker_stop_idx) == 0:
+            raise ValueError('Tracker {} has not stopped. All Trackers '
+                             'must be stopped.'.format(tracker_id))
+        responses = json.loads(raw[tracker_stop_idx[0]][2])['responses']
+        # feed in responses from tracker_ID_stop
+        [tr[-1].respond(r) for r in responses]
+    return tr
+
+
+def reconstruct_dealer(fname):
+    """Reconstruct TrackerDealer object from .tab files. The
+    ``reconstruct_tracker`` function will be called to retrieve the trackers.
+
+    Parameters
+    ----------
+    fname : str
+        Input filename.
+
+    Returns
+    -------
+    dealer : list of TrackerDealer
+        The TrackerDealer objects with all responses such that they are in
+        their stopped state.
+    """
+    from ..stimuli import TrackerDealer
+    raw = read_tab_raw(fname)
+
+    # find infor on dealer
+    dealer_idx = np.where([r[1] == 'dealer_identify' for r in raw])[0]
+    if len(dealer_idx) == 0:
+        raise ValueError('There are no TrackerDealers in this file.')
+    dealer = []
+    for ii in dealer_idx:
+        dealer_id = ast.literal_eval(raw[ii][2])['dealer_id']
+        dealer_init_str = 'dealer_' + str(dealer_id) + '_init'
+        dealer_dict_idx = np.where([r[1] == dealer_init_str
+                                    for r in raw])[0][0]
+        dealer_dict = ast.literal_eval(raw[dealer_dict_idx][2])
+        dealer_trackers = dealer_dict['trackers']
+
+        # match up tracker objects to id
+        trackers = reconstruct_tracker(fname)
+        tr_objects = []
+        for t in dealer_trackers:
```

Comment:
And are the trackers being stored in that shape within the dealer? The ``numpy.array`` of trackers needs to be ``reshape``d to that size.